File: server/domain/openapivo/network.go
package openapivo

type Network struct {
	ID                   uint   `json:"id"`
	Name                 string `json:"name"`
	CommunicationPeriod  uint   `json:"communicationPeriod"`
	CommunicationPeriod2 uint   `json:"communicationPeriod2"`
	CommunicationOffset  uint   `json:"communicationOffset"`
	GroupSize            uint   `json:"groupSize"`
	Mode                 uint8  `json:"mode"`
}

type NetworkDetail struct {
	ID                   uint   `json:"id"`
	Name                 string `json:"name"`
	CommunicationPeriod  uint   `json:"communicationPeriod"`
	CommunicationPeriod2 uint   `json:"communicationPeriod2"`
	CommunicationOffset  uint   `json:"communicationOffset"`
	GroupSize            uint   `json:"groupSize"`
	Mode                 uint8  `json:"mode"`

	Gateway NetworkDevice   `json:"gateway,omitempty"`
	Nodes   []NetworkDevice `json:"nodes"`
}

type NetworkDevice struct {
	MacAddress string `json:"macAddress"`
	Parent     string `json:"parent"`
}

File: src/types/network.ts
import { Device } from './device';
import { Asset } from './asset';
import intl from 'react-intl-universal';

export type Network = {
  id: number;
  name: string;
  gateway: Device;
  asset: Asset;
  routingTables: [];
  nodes: Device[];
  communicationPeriod: number;
  communicationPeriod2: number;
  communicationOffset: number;
  groupSize: number;
  mode: number;
  groupSize2: number;
  intervalCnt: number;
};

export enum NetworkProvisioningMode {
  Mode1 = 1,
  Mode2
}

export namespace NetworkProvisioningMode {
  export function toString(mode: NetworkProvisioningMode) {
    switch (mode) {
      case NetworkProvisioningMode.Mode2:
        return intl.get('WSN_MODE') + '2';
      default:
        return intl.get('WSN_MODE') + '1';
    }
  }
}
